streaming/vision/ade20k.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2022 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Callable, Optional, Tuple
+
+from torch import Tensor
+
+from streaming.base import Dataset
+
+
+class ADE20k(Dataset):
+    """
+    Implementation of the ADE20k dataset using streaming Dataset.
+
+    Args:
+        local (str): Local filesystem directory where dataset is cached during operation.
+        split (str): The dataset split to use, either 'train' or 'val'.
```

Comment:
split needs to remain Optional

when split is None, it is internally turned into the empty string, which os.path.joins to no split subdir and still correctly works

Reply:
while i doubt anyone would particularly want to make their own splits of for example ADE20k, it would be highly confusing to have this be inconsistent

Reply:
Make Sense. move it to Optional in https://github.com/mosaicml/streaming/pull/9/commits/6972e99881cba28a7035b300973d58e07a5b0509.